src/components/SearchBar.tsx

```diff
@@ -1,31 +1,40 @@
 import React from 'react';
-import { Input, InputGroup, InputLeftElement, Box } from '@chakra-ui/react';
+import {
+  Input,
+  InputGroup,
+  InputLeftElement,
+  Box,
+  VStack,
+} from '@chakra-ui/react';
 import { SearchIcon } from '@chakra-ui/icons';
 import { useColorMode } from '@chakra-ui/react';
-interface SearchBarProps {
-  value: string;
-  onChange: (value: string) => void;
+export interface SearchBarProps {
+  data: { value: string; onChange: (value: string) => void }[];
 }
 
-const SearchBar: React.FC<SearchBarProps> = ({ value, onChange }) => {
+const SearchBar: React.FC<SearchBarProps> = ({ data }) => {
   const { colorMode } = useColorMode();
   const bgColor = colorMode === 'dark' ? 'gray.700' : 'white';
   const borderColor = colorMode === 'dark' ? 'gray.600' : 'gray.200';
 
   return (
     <Box bg={bgColor} borderColor={borderColor} className="search">
-      <InputGroup>
-        <InputLeftElement
-          pointerEvents="none"
-          children={<SearchIcon color="gray.300" />}
-        />
-        <Input
-          type="text"
-          placeholder="Search"
-          value={value}
-          onChange={(e) => onChange(e.target.value)}
-        />
-      </InputGroup>
+      <VStack spacing={4}>
+        {data.map((input, index) => (
```

Comment:
Tutaj raczej też jak traktujemy to jako pojedynczy komponent to nie ma sensu korzystać mapy.